src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,72 @@
+import { useEffect, useState } from 'react';
+
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../services/api';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+export const TodoModal = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    if (currentTodo) {
+      getUser(currentTodo.userId)
```

Comment:
Ensure that the `getUser` function correctly fetches user data based on `currentTodo.userId`. If the API or the function is not correctly implemented, it may lead to errors or unexpected behavior.